### PR TITLE
feat: introduce UnimplementedHook to avoid authors having to define empty functions

### PR DESCRIPTION
**Pull request by skyerus**

### CODE REVIEW

File: pkg/openfeature/hooks.go [hunk 1]
```diff
@@ -65,3 +65,26 @@ func (h HookContext) ProviderMetadata() Metadata {
...
+// UnimplementedHook implements all hook methods with empty functions
+// Include UnimplementedHook in your hook struct to avoid defining empty functions
+// e.g.
+// type MyHook struct {
+//   UnimplementedHook
+// }
+type UnimplementedHook struct{}
+
+func (u UnimplementedHook) Before(hookContext HookContext, hookHints HookHints) (*EvaluationContext, error) {
+	return nil, nil
```
Comment (james-milligan) [inline pkg/openfeature/hooks.go:81]: I could be wrong - but will this set the `EvaluationContext` to `nil` before it is used for flag evaluation?

Comment (beeme1mr) [reply pkg/openfeature/hooks.go:81]: Could you add a test for this?

Comment (skyerus) [reply pkg/openfeature/hooks.go:81]: Incorrect. The resultant context is only used if not `nil`
Will add a test

**Comment by james-milligan** [comment]
looks good!

**Review by james-milligan — APPROVED** [review]

File: pkg/openfeature/hooks.go (hunk 1 above)
Comment (skyerus) [reply pkg/openfeature/hooks.go:81]: Added test

**Pull request merged**